src/Build/Graph/ProjectGraph.cs

```diff
@@ -22,21 +26,56 @@ internal sealed class ProjectGraph
         /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
         public ProjectGraph(string entryProjectFile)
         {
-            var graphNode =
-                new ProjectGraphNode(ProjectCollection.GlobalProjectCollection.LoadProject(entryProjectFile));
-            _projectNodes.Add(graphNode);
+            LoadGraph(entryProjectFile);
         }
 
         /// <summary>
-        /// Constructs a graph starting from the given project files.
+        /// Get an unordered collection of all project nodes in the graph.
         /// </summary>
-        /// <param name="entryProjectFiles">The project files to use as the entry points in constructing the graph</param>
-        /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
-        public ProjectGraph(IEnumerable<string> entryProjectFiles) { }
+        public IReadOnlyCollection<ProjectGraphNode> ProjectNodes => _allParsedProjects.Values;
+
+        private ProjectGraphNode CreateNewNode(string projectFilePath)
+        {
+            var graphNode =
+                new ProjectGraphNode(ProjectCollection.GlobalProjectCollection.LoadProject(projectFilePath));
```

Comment:
Should we use Project or ProjectInstance? Command line msbuild uses ProjectInstance. Only VS uses Project, which is a bit more expensive, and in certain cases might actually lead to different results (not super sure that's the case though)